llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -14036,10 +14084,22 @@ bool SLPVectorizerPass::vectorizeStores(ArrayRef<StoreInst *> Stores,
 
       unsigned Sz = 1 + Log2_32(MaxVF) - Log2_32(MinVF);
       SmallVector<unsigned> CandidateVFs(Sz);
+      auto VFsToFill = make_range(CandidateVFs.begin(), CandidateVFs.end());
```

Comment:
Why need to use VFsToFill, why CandidateVFs cannot be used directly?

Reply:
Why need to try it first?

Reply:
If we start with the regular power-of-2 VFs first, then in many cases we vectorize with a power-of-2 VF smaller than the non-power-of-2 one eagerly.

Reply:
Something like 
```
SmallVector<unsigned> CandidateVFs(Sz+(non-power-of-2-vf ? 1 : 0));
unsigned Size = MinVF;
for_each(reverse(CandidateVFs), [&](unsigned &VF) {
        VF = Size > MaxVF ? non-power-of-2-vf : Size;
        Size *= 2;
      });
```
?